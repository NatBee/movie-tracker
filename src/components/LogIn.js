import React, { Component } from 'react';
import { redirect, withRouter } from 'react-router-dom';
import { setUser } from '../actions/index';
import * as actions from '../actions/index';
import { userSignIn } from '../apiCall';
<<<<<<< HEAD
import { connect } from 'react-redux';
<<<<<<< HEAD
=======
import { setUser } from '../actions/index';
import * as actions from '../actions/index'
=======
import './LogIn.css'
>>>>>>> Clean up adjust movie card size and format LogIn.js
>>>>>>> Clean up adjust movie card size and format LogIn.js

class LogIn extends Component {
  constructor(props) {
    super(props);
    
    this.state ={
      email: '',
      password: '',
    }
  }

  handleChange = (e) => {
    e.preventDefault();
    const field = e.target.name;
    const value = e.target.value;

    this.setState({ [field]: value });
  }

  loginSubmit = async (e) => {
    e.preventDefault();

    const { email, password } = this.state
    const results = await userSignIn( email, password );
    const user = await results.data;

    if (!results) {
      alert('Oops! The email and password do not match');
    } else {
      this.props.setUser(user);
      this.props.history.push('/favorites')
    }
  }

  render() {

    return (
      <div className="log-in-form">
        <form onSubmit={ this.handleSubmit }>
          <input type="text" 
                 name="email" 
                 placeholder="email address" 
                 value={ this.state.email } 
                 onChange={ this.handleChange }/>
          <input type="password" 
                 name="password" 
                 placeholder="password" 
                 value={ this.state.password } 
                 onChange={ this.handleChange }/>
          <input type="submit" />
        </form>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setUser: (user) => dispatch(setUser(user)),
});

export default withRouter(connect(null, mapDispatchToProps)(LogIn));

