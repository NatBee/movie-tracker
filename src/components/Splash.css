.splash-page {
  background-image: url('../movie.jpg');
  width: 100%;
  height: 800px;
  background-repeat: no-repeat;
  background-size: cover;
  margin-top: -55px;
}