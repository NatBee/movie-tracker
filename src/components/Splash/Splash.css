.splash-page {
  background-image: url('../../assets/movie.jpg');
  width: 100%;
  height: 800px;
  background-repeat: no-repeat;
  background-size: cover;
  margin-top: -55px;
}