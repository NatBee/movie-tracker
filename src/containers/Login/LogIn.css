input {
  width: 300px;
  padding: 10px;
  font-size: 1rem;
  display: block;
  margin: 15px auto;
}

.submit-button {
  padding: 10px;
  width: 322px;
  font-size: 1rem;
  border-radius: 0;
  margin-bottom: 15px;
}

.submit-button hover {
  border: none;
}

form {
  margin: auto;
  width: 350px;
  height: 300px;
  padding: 40px;
  background-color: black;
  opacity: 0.75;
  position: relative;
  top: 50px;
}

p{
  color: white;
}

.login-error {
  color: red;
  font-size: .75em;
  padding: 0 0 10px;
}

.acct-create {
  color: red;
  font-size: .75em;
  padding: 0 0 10px;
}