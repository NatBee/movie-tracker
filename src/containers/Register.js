import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setUser} from '../actions/index';
import { userSignIn, addNewUser } from '../apiCall'



class Register extends Component {
  constructor(props) {
    super(props);
    
    this.state ={
      name: '',
      email: '',
      password: '',
    }
  }

  handleChange = (e) => {
    e.preventDefault();
    const field = e.target.className;
    const value = e.target.value;

    this.setState({ [field]: value });
  }

  registerSubmit = async (e) => {

    const { name, email, password } = this.state

    e.preventDefault();
    const results = await addNewUser(name, email, password);
    const userResults = await userSignIn(email, password);
    const user = await userResults.data
    this.props.setUser(user)
  }

  render() {
    return (
      <div className="log-in-form">
        <form onSubmit={this.registerSubmit}>
          <input type="text" className="name" placeholder="Your Name" name="email" onChange={this.handleChange}/>
          <input type="text" className="email" placeholder="email address" name="email" onChange={this.handleChange}/>
          <input type="password" className="password" placeholder="password" name="password" onChange={this.handleChange}/>
          <input type="submit" />
        </form>
      </div>
    )
  }
}

// export const mapStateToProps = (store) => ({
//   name: store.user.name
// })

export const mapDispatchToProps = (dispatch) => ({
  addNewUser: (user) => dispatch(addNewUser(user)),
})

export default connect(null, mapDispatchToProps)(Register)


