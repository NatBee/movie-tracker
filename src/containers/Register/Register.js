import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, Link } from 'react-router-dom';
import { setUser } from '../../actions/index';
import { userSignIn, addNewUser } from '../../helper/apiCall';
import PropTypes from 'prop-types';

export class Register extends Component {
  constructor(props) {
    super(props);
    
    this.state ={
      name: '',
      email: '',
      password: '',
      errorMessage: null,
    }
  }

  handleChange = (e) => {
    e.preventDefault();
    const field = e.target.className;
    const value = e.target.value;

    this.setState({ [field]: value });
  }

  registerSubmit = async (e) => {
    e.preventDefault();
    
    const { name, email, password } = this.state
    const results = await addNewUser({ name, email, password });
    this.updateMessage(results)
  }

  updateMessage = async (results) => {
    if(results.status !== 'success') {
      this.setState({errorMessage: "That email is already registered. Please try again or login."})
    } else {
      this.autoSignIn();
    }
  }

  autoSignIn = async () => {
    const userEmail = this.state.email;
    const userPassword = this.state.password;

    const userResults = await userSignIn(userEmail, userPassword);
    const user = await userResults.data; 
    this.props.setUser(user)
    this.props.history.push('/')
  }

  render() {
    return (
      <div className="log-in-form">
        <form onSubmit={ this.registerSubmit }>
          <h3 className="acct-create">{ this.state.newAcctMessage }</h3>
          <input type="text" className="name" placeholder="Your Name" name="email" onChange={this.handleChange}/>
          <input type="text" className="email" placeholder="email address" name="email" onChange={this.handleChange}/>
          <input type="password" className="password" placeholder="password" name="password" onChange={this.handleChange}/>
          <button type="submit"
                  className="submit-button">Register</button>
          <h3 className="login-error">{ this.state.errorMessage }</h3>
          <Link to="/login">Login</Link>
        </form>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addNewUser: (user) => dispatch(addNewUser(user)),
  setUser: (user) => dispatch(setUser(user))
})

Register.propTypes = {
  history: PropTypes.object,
  addNewUser: PropTypes.func,
  setUser: PropTypes.func,
  user: PropTypes.object,
  registerSubmit: PropTypes.func,
  updateMessage: PropTypes.func,
  autoSignIn: PropTypes.func,
}

export default withRouter(connect(null, mapDispatchToProps)(Register));