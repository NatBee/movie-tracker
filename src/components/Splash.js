import React from 'react';
import LogIn from './LogIn';
import './Splash.css';

const Splash = () => {
  return (
    <div className="splash-page">
      { <LogIn /> }
    </div>
  )
}

export default Splash;