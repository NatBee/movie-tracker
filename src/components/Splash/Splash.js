import React, { Component } from 'react';
import LogIn from '../../containers/LogIn/LogIn';
import Register from '../../containers/Register/Register'
import './Splash.css';
import PropTypes from 'prop-types';

class Splash extends Component {
  renderComponent = () => {
    const route = this.props.location.pathname;
    
    return route === "/login" ? <LogIn /> : <Register />
  }

  render() {

    return (
      <div className="splash-page">
       { this.renderComponent() }
      </div>
    )
  }
}

Splash.propTypes = {
  location: PropTypes.object,
}

export default Splash;