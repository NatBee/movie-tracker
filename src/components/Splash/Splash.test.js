/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import Splash from './Splash';
import LogIn from '../../containers/LogIn/LogIn';
import Register from '../../containers/Register/Register';

describe('Splash', () => {
  let wrapper;

 it('should render the LogIn component', () => {
    const mockLocation = {
      pathname: '/login'
    }
    wrapper = shallow(<Splash location={ mockLocation }/>);

    expect(wrapper.find(LogIn).length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

 it('should render the Register component', () => {
  const mockLocation = {
      pathname: '/register'
    }
    wrapper = shallow(<Splash location={ mockLocation }/>);

    expect(wrapper.find(Register).length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
 });
})