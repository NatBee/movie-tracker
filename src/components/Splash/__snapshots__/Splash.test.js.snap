// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash should render the LogIn component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Splash
    location={
      Object {
        "pathname": "/login",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <withRouter(Connect(LogIn)) />,
      "className": "splash-page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <withRouter(Connect(LogIn)) />,
        "className": "splash-page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Splash should render the Register component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Splash
    location={
      Object {
        "pathname": "/register",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <withRouter(Connect(Register)) />,
      "className": "splash-page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <withRouter(Connect(Register)) />,
        "className": "splash-page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
